aiohttp_sentry/__init__.py

```diff
@@ -1,22 +1,58 @@
+import asyncio
 from functools import partial
+import sys
 
 import raven
 import raven_aiohttp
 
 
 class SentryMiddleware:
 
-    def __init__(self, sentry_kwargs=None):
+    def __init__(self, sentry_kwargs=None, *, install_excepthook=True, loop=None):
         if sentry_kwargs is None:
             sentry_kwargs = {}
 
         sentry_kwargs = {
             'transport': raven_aiohttp.AioHttpTransport,
             'enable_breadcrumbs': False,
+            # by default, do not let raven.Client install its own excepthook
+            'install_sys_hook': not install_excepthook,
             **sentry_kwargs,
         }
         self.client = raven.Client(**sentry_kwargs)
 
+        if install_excepthook:
+            self.update_excepthook(loop)
+
+    def update_excepthook(self, loop=None):
+        """Update sys.excepthook so that it closes the Sentry transport.
+
+        If a custom event loop is provided,
+        it will be used for closing the Sentry transport
+        derived from raven_aiohttp.AioHttpTransportBase
+        instead of the event loop for the current context
+        defined by the current event loop policy.
```

Comment:
Are you really using [semantic newlines](http://rhodesmill.org/brandon/2012/one-sentence-per-line/)? 😍

I love it :D

Reply:
I have not heard about the term yet. The simplicity of editing is definitely useful. It was not my original intention, but I agree that striving to achieve it is a good practice. I try to break the lines at points between the sentences, ideas or other semantic units, so that it is easier to read and understand.